server/route.go

```diff
@@ -351,6 +351,16 @@ func (c *client) sendConnect(tlsRequired bool) {
 
 // Process the info message if we are a route.
 func (c *client) processRouteInfo(info *Info) {
+	// We may need to update route permissions and will need the account
+	// sublist. Since getting the account requires server lock, do the
+	// lookup now.
+
+	// FIXME(dlc) - Add account scoping.
+	gacc := c.srv.globalAccount()
```

Comment:
Safe to reference c.srv without capturing it first under a client lock?

Reply:
Yes. This is assigned when client structure is created and never set to nil. Remember, I had argued that we should fix the tests that create client object directly (so possibly without c.srv value) than have the whole server code alway checking for c.srv != nil.

Reply:
which then makes you wonder if it is possible to be nil and does it need to be captured under lock :-)

Reply:
we should probably do that then longer term.